src/main/java/it/polimi/telco/model/Product.java

```diff
@@ -0,0 +1,55 @@
+package it.polimi.telco.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "product", schema = "telco")
+@NamedQuery(name = "Product.getProductById", query = "SELECT p FROM Product p  WHERE p.id = ?1 ")
```

Comment:
I think with EntityManager is easy to retrieve an object by its id without Named Queries
The code should look like this
```java
public Project findProjectById(int projectId) {
    return (em.find(Project.class, projectId)); // em is entity manager
}
```